browser/brave_vpn/win/brave_vpn_wireguard_service/service/brave_wireguard_manager.cc

```diff
@@ -6,19 +6,69 @@
 #include "brave/browser/brave_vpn/win/brave_vpn_wireguard_service/service/brave_wireguard_manager.h"
 
 #include <string>
+
+#include "base/base64.h"
 #include "base/logging.h"
 #include "base/strings/utf_string_conversions.h"
 #include "brave/browser/brave_vpn/win/brave_vpn_wireguard_service/service/wireguard_tunnel_service.h"
+#include "brave/components/brave_vpn/common/wireguard/wireguard_utils.h"
 
 namespace brave_vpn {
 
-HRESULT BraveWireguardManager::EnableVpn(const wchar_t* config,
+HRESULT BraveWireguardManager::EnableVpn(BSTR public_key,
+                                         BSTR private_key,
+                                         BSTR address,
+                                         BSTR endpoint,
                                          DWORD* last_error) {
-  if (!config || !last_error) {
-    VLOG(1) << "Invalid parameters";
+  if (!last_error) {
+    VLOG(1) << "last_error must be set";
+    return E_FAIL;
+  }
+
+  // if all params are empty, reconnect using last known good config.
+  // browser/brave_vpn/win/brave_vpn_wireguard_service/service/wireguard_tunnel_service.cc
+  bool reconnect_using_last_config = public_key && wcslen(public_key) == 0 &&
+                                     private_key && wcslen(private_key) == 0 &&
+                                     address && wcslen(address) == 0 &&
+                                     endpoint && wcslen(endpoint) == 0;
+  if (reconnect_using_last_config) {
+    *last_error = !brave_vpn::wireguard::LaunchWireguardService(L"");
```

Comment:
do we expect this to happen under some circumstances or are we trying to handle an unexpected condition here? If it's the latter I think we should just fail here because we might be unintentionally opening another attack vector

Reply:
This is the current behavior - I thought I made comments in this PR explaining... but basically the tray program is passing empty string for the arguments. This triggers the "reconnect using the last known config" logic.

We could rework the tray program - but that's a bigger change of course.

Reply:
I think, best is to harden this part with the follow-up in https://github.com/brave/brave-browser/issues/39229